File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int left = 0, right = 0;
        
        for (auto ch : s){
            if (ch == '(')
                left += 1;

            else if (ch == ')'){
                if (left > 0)
                    left -= 1;

                else
                    right += 1;
            }
        }
        
        return left + right;
    }
};

File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        left = right = 0
        
        for ch in s:
            if ch == '(':
                left += 1

            elif ch == ')':
                if left > 0:
                    left -= 1

                else:
                    right += 1
        
        return left + right